packages/device-connector/src/deployment-server/connection.ts

```diff
@@ -39,23 +41,24 @@ const readConnectorData = async (): Promise<ConnectorData> => {
 }
 
 const generateConnectorData = async (): Promise<ConnectorData> => {
-  const registrationResponse = await fetch(`http://${address}/api/connectors`, {
+  const registrationResponse = await fetch(`${deployUri}/connectors`, {
     method: 'POST'
   })
-  const { id } = await registrationResponse.json() as any
 
-  return await new Promise<ConnectorData>((resolve, reject) => {
-    if (id === '') {
-      reject(new Error('No valid id response'))
-    }
+  if (!registrationResponse.ok) {
+    throw httpError(registrationResponse)
+  }
 
-    const connectorData: ConnectorData = {
-      id: id,
-      uri: address
-    }
+  const { id } = await registrationResponse.json() as any
```

Comment:
```suggestion
  const { id } = await registrationResponse.json() as {id: string}
```

Reply:
ah thanks alot. Was wondering how to set the type here